src/compiler/crystal/syntax/lexer.cr

```diff
@@ -618,6 +644,18 @@ module Crystal
           else
             @token.type = :"&*"
           end
+        when '/'
+          case next_char
+          when '/'
+            case next_char
```

Comment:
ditto

Reply:
With `if`s it would be (start at line 648, replacing the 2 `case`s):
```crystal
raise "&/ is not a valid operator. Did you mean &// ?" if next_char != '/'
if next_char == '='
  next_char :"&//="
else
  @token.type = :"&//"
end
```

Reply:
At a note, the previous code was:
```crystal
case next_char
when '/'
  case next_char
  when '='
    next_char :"&//="
  else
    @token.type = :"&//"
  end
else
  raise "&/ is not a valid operator. Did you mean &// ?"
end
```